engine/source/elements/thickshell/solide8c/s8cfint_reg.F

```diff
@@ -0,0 +1,937 @@
+     9   PZ4,     PZ5,     PZ6,     PZ7,
+     A   PZ8,     IMAT,    ITASK,   DT2T,
+     B   VOL0,    NFT,     NLAY,    NPTR,
+     C   NPTS,    IR,      IS,      WS,
+     D   AS  ,    ICR,     ICS,     ICT,
+     E   BUFNLTS, AREA,    NODADT,  DT1, 
+     F   DT12,    DTFAC1,  DTMIN1,  IDTMIN)
+C-----------------------------------------------
+C   M o d u l e s
+C----------------------------------------------- 
+      USE NLOCAL_REG_MOD
+      USE ELBUFDEF_MOD
+C-----------------------------------------------
+C   I m p l i c i t   T y p e s
+C-----------------------------------------------
+#include      "implicit_f.inc"
+C-----------------------------------------------
+C   G l o b a l   P a r a m e t e r s
+C-----------------------------------------------
+#include      "parit_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER, INTENT(IN) :: NFT,NLAY,NPTR,NPTS,IR,IS,
+     .                       NEL,IMAT,ITASK,ICR,ICS,ICT
+      INTEGER, DIMENSION(NEL) :: NC1,NC2,NC3,NC4,NC5,NC6,NC7,NC8
+      my_real, DIMENSION(NEL), INTENT(IN) :: 
+     .   OFFG
+      my_real, INTENT(INOUT) ::
+     .   DT2T
+      my_real, DIMENSION(NEL), INTENT(IN) :: 
+     .   VOL,VOL0,AREA
+      TYPE(NLOCAL_STR_), TARGET :: NLOC_DMG 
+      my_real, DIMENSION(9,9), INTENT(IN) :: 
+     .   WS,AS
+      my_real, DIMENSION(NEL,NPTR*NPTS*NLAY), INTENT(INOUT) :: 
+     .   VAR_REG,PX1,PX2,PX3,PX4,PX5,PX6,PX7,PX8,
+     .   PY1,PY2,PY3,PY4,PY5,PY6,PY7,PY8,PZ1,PZ2,
+     .   PZ3,PZ4,PZ5,PZ6,PZ7,PZ8
+      TYPE(BUF_NLOCTS_), INTENT(INOUT), TARGET :: BUFNLTS
+      my_real, INTENT(IN) ::
+     .   DT1,DT12
+      INTEGER, INTENT(IN) :: NODADT,IDTMIN(102)
+      my_real, INTENT(IN) :: DTFAC1(102),DTMIN1(102)
+C-----------------------------------------------
+C   L o c a l   V a r i a b l e s
+C-----------------------------------------------
+      INTEGER I,J,II,K,NNOD,N1,N2,N3,N4,N5,N6,N7,N8,
+     .        L_NLOC,IP,NDDL,NDNOD
+      my_real 
+     . L2,XI,B1,B2,B3,B4,B5,B6,B7,B8,H(8),
+     . A1,A2,A3,A4,A5,A6,A7,A8,
+     . C1,C2,C3,C4,C5,C6,C7,C8,
+     . ZETA,SSPNL,DTNL,LE_MAX,MAXSTIF,
+     . ZR,ZS,ZT,PR(8),PS(8),PT(8),WI,
+     . BTH1,BTH2,NTH1,NTH2,DT2P,DTNOD,
+     . K1,K2,K12,DTNL_TH
+      my_real, DIMENSION(NEL) :: 
+     . LC,THK,LTHK
+      my_real, DIMENSION(NEL,NLAY) :: 
+     . F1,F2,F3,F4,F5,F6,F7,F8
+      my_real, DIMENSION(:,:) ,ALLOCATABLE ::
+     . STI1,STI2,STI3,STI4,STI5,STI6,STI7,STI8
+      my_real, DIMENSION(:) ,ALLOCATABLE   :: 
+     . BTB11,BTB12,BTB13,BTB14,BTB15,BTB16,BTB17,BTB18,
+     . BTB22,BTB23,BTB24,BTB25,BTB26,BTB27,BTB28,BTB33,
+     . BTB34,BTB35,BTB36,BTB37,BTB38,BTB44,BTB45,BTB46,
+     . BTB47,BTB48,BTB55,BTB56,BTB57,BTB58,BTB66,BTB67,
+     . BTB68,BTB77,BTB78,BTB88
+      INTEGER, DIMENSION(:), ALLOCATABLE   ::
+     . POS1,POS2,POS3,POS4,POS5,POS6,POS7,POS8
+      my_real, POINTER, DIMENSION(:) :: 
```

Comment:
@vdavaze 

Can you use ALLOCATABLE array in place of array ?
( it is not recommended to use POINTER if we have other possible solution)

Thanks,
Sébastien

Reply:
Hi @sebastienVilleneuve 

The code has been fixed :-) . Pointers have been removed from the code. 

Thanks a lot. 
Valentin.